lib/rexml/parsers/baseparser.rb

```diff
@@ -335,6 +335,10 @@ def pull_event
               @nsstack.shift
               last_tag = @tags.pop
               md = @source.match( CLOSE_MATCH, true )
+              if md && !last_tag
```

Comment:
Could you use `and`?

Reply:
Definitely. Sorry I missed using `and` which is commonly used in this project.
I rebase'ed to use `and` instead of `&&` at my first commit.